File: leetcode/array_string/common_divisor_1071.java
class Solution {
    public String gcdOfStrings(String str1, String str2) {
        if (!(str1 + str2).equals(str2 + str1))
            return "";
        int a = str1.length(), b = str2.length();
        int result = 0;
        int temp = 0;
        while (b > 0) {
            temp = b;
            b = a % b;
            a = temp;
            result = a;
        }
        return str1.substring(0, result);

    }
}

// if (str1 + str2 != str2 + str1) return "";
// 這會比較到Java 中使用 == 或 != 比較字串，會比的是記憶體位址（reference equality），不是字串內容。

File: leetcode_75/array_string/common_divisor_1071.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
            if str1 + str2 != str2 + str1:
                return ""
            a, b = len(str1), len(str2)
            result = 0
            while b:
                 a, b = b, a%b
                 result = a
            return str1[:result]

    def gcdOfStrings(str1: str, str2: str) -> str:
        if str1 + str2 != str2 + str1:
            return ""
        from math import gcd
        return str1[:gcd(len(str1), len(str2))]



tt = Solution()
tt.gcdOfStrings("aaa", "ABCABCABC")